service/key/key.go

```diff
@@ -119,8 +119,30 @@ func IsCAPIManagementCluster(provider cluster.Provider) bool {
 }
 
 func ClusterProvider(cluster metav1.Object, provider cluster.Provider) string {
-	if IsEKSCluster(cluster) {
-		return "eks"
+	// We keep the existing behavior for vintage management clusters
+	if !IsCAPIManagementCluster(provider) {
+		return provider.Kind
+	}
+
+	if c, ok := cluster.(*capi.Cluster); ok {
+		switch c.Spec.InfrastructureRef.Kind {
+		case "AWSCluster":
+			return "capa"
+		case "AWSManagedCluster":
+			return "eks"
+		case "AzureCluster":
+			return "capz"
+		case "AzureManagedCluster":
+			return "aks"
+		case "VCDCluster":
+			return "cloud-director"
+		case "VSphereCluster":
+			return "vsphere"
+		case "GCPCluster":
+			return "gcp"
+		case "GCPManagedCluster²":
+			return "gke"
+		}
 	}
 	return provider.Kind
```

Comment:
What are the cases where we could fall back to this default behaviour? I mean, is it expected that some clusters would not have a `provider.Kind`?
If this is not expected, maybe we should rather set `Unknown` so we can detect this more easily?

Reply:
So it could technically happen for vintage or if that value was not set by the customer so erroring out makes sense :)